cypher-shell/src/main/java/org/neo4j/shell/CommandReader.java

```diff
@@ -0,0 +1,41 @@
+package org.neo4j.shell;
+
+import jline.console.ConsoleReader;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CommandReader {
+    private final ConsoleReader reader;
+    private final Shell shell;
+    static final Pattern MULTILINE_BREAK = Pattern.compile("\\\\\\s*$");
```

Comment:
We want to catch a single backslash, followed by whitespace to the end of the line.

This is what it needs to look like. We have actually tested several variants and this does do what we expect. It does look like there is a backslash too many, but there actually isn't.